File: Testing/Stdlib Test Code/test_object_out.py
from dataclasses import dataclass, field

class Animal:
    def __init__(self, name="", age=0):
        self.name = name
        self.age = age
    def speak(self):
        return '...'

class Dog(Animal):
    def __init__(self, name="", age=0, breed=""):
        super().__init__(name, age)
        self.breed = breed
    def speak(self):
        return 'Woof!'
    def info(self):
        return self.breed


def main():
    d = Dog()
    d.name = 'Rex'
    d.age = 3
    d.breed = 'Labrador'
    print(d.name)
    print(d.age)
    print(d.speak())
    print(d.info())
    return 0


if __name__ == "__main__":
    main()

File: Testing/Stdlib Test Code/test_object_out.js
class Animal {
    constructor(name = "", age = 0) {
        this.name = name;
        this.age = age;
    }
    speak() {
        return "...";
    }
}

class Dog extends Animal {
    constructor(name = "", age = 0, breed = "") {
        super(name, age);
        this.breed = breed;
    }
    speak() {
        return "Woof!";
    }
    info() {
        return this.breed;
    }
}

function main() {
    let d = new Dog();
    d.name = "Rex";
    d.age = 3;
    d.breed = "Labrador";
    console.log(d.name);
    console.log(d.age);
    console.log(d.speak());
    console.log(d.info());
    return 0;
}


main();
